src/components/TableDialog.vue

```diff
@@ -60,12 +115,61 @@ export default Vue.extend({
           displayName: 'CSV',
         },
       ] as FileType[],
+      file: null as File | null,
+      fileName: null as string | null,
+      fileUploadError: null as string | null,
+      tableCreationError: null as string | null,
+      key: undefined as string | undefined,
+      overwrite: false,
     };
   },
+  computed: {
+    createDisabled(): boolean {
+      return (
+        !this.file ||
+        !this.fileName ||
+        !!this.fileUploadError
+      );
+    },
+  },
   methods: {
-    uploadSuccess() {
-      this.tableDialog = false;
-      this.$emit('success');
+    handleFileInput(file: File) {
+      this.file = file;
+
+      if (!file) {
+        this.fileUploadError = null;
+      } else if (!validFileType(file, this.types)) {
+        this.fileUploadError = 'Invalid file type';
+      } else {
+        this.fileName = this.fileName || getFileName(file);
+        this.fileUploadError = null;
+      }
+    },
+    async createTable() {
+      const {
+        file,
+        workspace,
+        fileName,
+        key,
+        overwrite,
+      } = this;
+
+      if (file === null || fileName === null) {
+        return;
+      }
```

Comment:
Is this necessary since it's already checked by `createDisabled`?

Reply:
Good point, I'll update it.

Reply:
Actually, while you're correct, the reason I removed it is because typescript doesn't know that, and complains that `file` or `fileName` could be `null`, so I've left it as is.